include/openPMD/IO/ADIOS/ADIOS2Auxiliary.hpp

```diff
@@ -0,0 +1,98 @@
+#pragma once
+
+#include "openPMD/config.hpp"
+#if openPMD_HAVE_ADIOS2
+#include "openPMD/Datatype.hpp"
+#include <adios2.h>
+#include <utility>
+#include <vector>
+
+namespace openPMD
+{
+namespace detail
+{
+    // ADIOS2 does not natively support boolean values
```

Comment:
I think we don't use it yet. I have used it in my `topic-streaming` branch to indicate whether an iteration has been finalized, but I noticed that it is not so straightforward to do so. Since `bool`s are coerced to `unsigned char`, you need to [pay attention](https://github.com/franzpoeschel/openPMD-api/blob/909fa5efcd6a692535aa1cee4f905b6d439b7b24/src/IO/ADIOS/ADIOS2IOHandler.cpp#L686) when reading whether an attribute was `bool` or whether it was genuinely an `unsigned char`. If I remember correctly, the other backends don't handle this case yet. My assumption would be that this hasn't come up yet since bools have not been used?
We might think about removing `BOOLEAN` alltogether from `openPMD::Datatypes`?

Reply:
It's probably not a bad idea, let's document this in an issue.